DataStorageEngine/DSECommonSenseProxy.h

```diff
@@ -0,0 +1,172 @@
+ 
+	
+	The DSECommonSenseProxy class does not now about DataStorageEngine concepts, it simply wraps the API that is available for CommonSense. Any translations between concepts in the DataStorageEngine and CommonSense should be made by other objects (most notably the difference between Source and Device). 
+ 
+	The DSECommonSenseProxy uses a Session ID and Application Key to get authorized for API access. These parameters are managed automatically but can be set by others as well. -->> do we want this?
+ 
+ */
+@interface DSECommonSenseProxy : NSObject
+
+@property NSString* sessionID;		//Do we want the proxy to manage the session ID? If we don't it has to be provided to every method.
+@property NSString* applicationKey; //Do we want the proxy to manage the app key? If we don't it has to be provided to every method.
+
+#pragma mark User functions
+/**
+ @name Users
+ */
+
+/**
+ Registers a new user as an account to CommonSense
+
+ @note Note that when the user is registered, the user is not yet logged in. This should be done after the registeration succeeds by the user/developer.
+ 
+ @param username		A user account in commonsense is uniquely identified by a username. It can be any string, however it cannot exist yet in the commonsense account database. The call will fail if the user account already exists.
+ @param password		A password in commonsense does not have any specific requirements. It will be MD5 hashed before sending to the server so the user does not have to provide a hashed password.
+ @param email			Optional, an email address to link to the account
+ @param successCallback A callback method that will be called in the case the API call succeeds. It does not receive any input parameters.
+ @param failureCallback A callback method that will be called in case of failing the API call. It will receive an NSError object with details about the cause of the failure.
+ */
++ (void) registerUser: (NSString *) username andPassword: (NSString *) password andEmail: (NSString *) email andSuccessCallback:(void(^)()) successCallback andFailureCallback:(void(^)(NSError* error)) failureCallback;
+
+/**
+ Login a user
+ 
+ @note If a user is already logged in this call will fail. That user should be logged out first before a new login attempt can be made. The failureCallback will be called in that case.
+ 
+ @param username		A user account in commonsense is uniquely identified by a username.
+ @param password		A password in commonsense does not have any specific requirements. It will be MD5 hashed before sending to the server so the user does not have to provide a hashed password.
+ @param successCallback A callback method that will be called in the case the API call succeeds. It does not receive any input parameters.
```

Comment:
Hard to copy these kind of anonymous callback functions in JAVA. I can see how this works pretty smooth in objective-c, but I don't think this is possible in Android.